base/array.jl

```diff
@@ -1111,16 +1111,25 @@ function _growbeg!(a::Vector, delta::Integer)
     @_noub_meta
     delta = Int(delta)
     delta == 0 && return # avoid attempting to index off the end
-    delta >= 0 || throw(ArgumentError("grow requires delta >= 0"))
+    delta ≥ 0 || throw(ArgumentError("grow requires delta ≥ 0"))
     ref = a.ref
     mem = ref.mem
     len = length(a)
     offset = memoryrefoffset(ref)
+    if len == 0
+        # Fast path: no data to shift, just shift offset if space permits
+        offset ≥ delta || throw(ArgumentError("requested prepend exceeds capacity"))
```

Comment:
this error is wrong. If there isn't sufficient space, this should be an allocation.